core/src/builder/core.rs

```diff
@@ -163,16 +163,20 @@ where
 	pub fn commit_storage_changes(
 		&mut self,
 		changes: StorageChanges<B::State, Block>,
+		at: BlockId::Hash,
 	) -> Result<(), ()> {
-		let at = BlockId::Hash(self.client.info().best_hash);
 		let mut op = self.backend.begin_operation().unwrap();
+
 		self.backend.begin_state_operation(&mut op, at).unwrap();
+
 		let info = self.client.info();
+
 		if info.best_hash == info.finalized_hash {
```

Comment:
This seems a bit off to me, mainly because we are doing more than committing something to storage. At a first glance, this looked like some sort of undocumented behaviour that changes state.